src/VisualStudio/Core/Def/Implementation/VisualStudioSupportsFeatureService.cs

```diff
@@ -31,13 +33,38 @@ public VisualStudioTextBufferSupportsFeatureService()
             }
 
             public bool SupportsCodeFixes(ITextBuffer textBuffer)
-                => SupportsCodeFixesWorker(GetContainedDocumentId(textBuffer));
+            {
+                // Disable code fix entry points when running under the LSP editor.
+                // The LSP client will handle displaying lightbulbs.
+                if (textBuffer.IsInLspEditorContext())
```

Comment:
i really don't love this sort of thing.  @jasonmalinowski if he has better ideas for how we can light this sort of thing up without having to leak information about Lsp into all these different contexts.

Here's what i don't get:

1. if we are enabling LSP for thsi stuff, i woudl expect roslyn wouldn't have to do much of anything.  It's just that the editor/platform will see taht they can access the LSP side of things and use that.  They can then choose to use that, or call into their existing non-lsp api.  From our perspective, we would not care.

Reply:
It's a little problematic
Basically the LSP client command handlers will be active if we provide the correct server capabilities for the feature.  However, their command handlers are not ordered in regards to ours, so it is somewhat undeterministic which handler will be called if both return that they are available.  Additionally for command handlers that aggregate the results, we would get duplicates unless we specifically disable ours.

Reply:
So in this case it isn't a matter of command handlers, but the ISuggestedAction editor API. If we enable LSP, the editor still has to call ISuggestedAction APIs as well for anybody _else_ using those, and there's no way for them to know which one is ours to not call it unless they have some other MEF metadata or something.

That all said, @dibarbet I would have expected these sorts of checks to actually be in the ISuggestedAction code, since how is this working for other workspaces like interactive or misc files?

Reply:
@jasonmalinowski this service is used by misc files.  for interactive I don't think we support rename/code actions at all there right?

I originally had these in the suggested action source, but this seemed the right place to check

Reply:
I guess I'd be expecting the LSP code to be calling the supports feature service itself to figure out whether it should be offering fixes or something like that. But then again I guess that's only text-buffer based; there's also a document-based equivalent service that seems...suspect.

> for interactive I don't think we support rename/code actions at all there right?

https://github.com/dotnet/roslyn/blob/c510a1171e067fd9b3d52f8e53cb79b823f27b3e/src/EditorFeatures/Core.Wpf/Interactive/InteractiveSupportsFeatureService.cs#L42-L43

So _sometimes_. Now we're not using LSP there yet, but not sure if this could cause issues.

> @jasonmalinowski this service is used by misc files.

Ah, right.

Reply:
Right yeah it doesn't call the supports feature service since it only applies to text buffers.

However moving this to the code action entry point seems to make sense to me as it would 
1.  cover all the various workspaces
2.  if we did modify the service to be document based and LSP started using it then stuff would work appropriately w/out changes.

With that in mind, I guess now I'm leaning towards moving it back to the suggested action entry point,.

Reply:
moved the check into the source provider